tests/agent/test_event_handler.py

```diff
@@ -18,20 +31,46 @@
 pytestmark = pytest.mark.unit
 
 
-def test_event_handler_raises_for_unknown_event():
+def test_event_handler_unknown_event():
     event = UnknownEvent()
     correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
     context = MagicMock(autospec=CloudDataContext)
     handler = EventHandler(context=context)
+    result = handler.handle_event(event=event, id=correlation_id)
+    assert result.type == "unknown_event"
+
+
+@pytest.mark.parametrize(
+    "event_class,action_class",
+    {
+        RunOnboardingDataAssistantEvent: RunOnboardingDataAssistantAction,
+        RunMissingnessDataAssistantEvent: RunMissingnessDataAssistantAction,
+        ListTableNamesEvent: ListTableNamesAction,
+        RunCheckpointEvent: RunCheckpointAction,
+        RunColumnDescriptiveMetricsEvent: ColumnDescriptiveMetricsAction,
+        DraftDatasourceConfigEvent: DraftDatasourceConfigAction,
+    }.items(),
+)
+def test_get_action(event_class, action_class):
+    event = MagicMock(spec=event_class)
+    context = MagicMock(autospec=CloudDataContext)
+    handler = EventHandler(context=context)
+    action_class_actual = handler.get_event_action(event)
+    assert isinstance(action_class_actual, action_class)
+
 
-    with pytest.raises(UnknownEventError):
-        handler.handle_event(event=event, id=correlation_id)
+def test_event_action_unknown():
+    context = MagicMock(autospec=CloudDataContext)
+    handler = EventHandler(context=context)
+    action = handler.get_event_action(UnknownEvent())
+    assert isinstance(action, UnknownEventAction)
 
 
 def test_event_handler_handles_run_missingness_data_assistant_event(mocker):
-    action = mocker.patch(
-        "great_expectations_cloud.agent.event_handler.RunMissingnessDataAssistantAction"
-    )
+    mock_action = MagicMock(autospec=RunMissingnessDataAssistantAction)
+    # Override with mock
+    EventHandler._EVENT_TO_ACTION_MAP[RunMissingnessDataAssistantEvent] = mock_action
```

Comment:
You don't need to use `MagicMock` for this.
You can & should use the `mocker` fixture because it automatically remove the patches at the end of the test.

https://pytest-mock.readthedocs.io/en/latest/usage.html
```python
def test_event_handler_handles_run_missingness_data_assistant_event(mocker):
    action = mocker.patch(
        "great_expectations_cloud.agent.event_handler.RunMissingnessDataAssistantAction", autospec=RunMissingnessDataAssistantAction
    )

```

Reply:
I do now see that overriding with the mock affects subsequent tests. However, mocker.patch this did not work because class attribute `EventHandler._EVENT_TO_ACTION_MAP` gets loaded before the mock is set. 

When I use the patch you provided, I get the real object and test fails.

Check out this use of mocker against the map: b422df3f15a937ccc91107f79fbb7c5270fd64d7

Reply:
Looking at that commit it looks like the tests passed.
`EventHandler._EVENT_TO_ACTION_MAP` being defined before the mock shouldn't matter.
Can you link to the failure?

Reply:
I've created a `PR` to provide some guidance around this with the `ruff` `banned-api` rule.
- https://github.com/great-expectations/cloud/pull/158

Reply:
If for some reason this still doesn't work we can do an inline `noqa: TID251 # comment with explanation` ignore

Reply:
FYI I did a refactor of most of these tests here in case it's helpful: https://github.com/great-expectations/cloud/pull/150/files#diff-73699a97bab49c907e00515541bf18c84ff213b90545dd6f139fcdafd3ecc7b3

Reply:
fyi going to pull in Anthony's tests which dont have this testing issue. Once merged into main, ill pull into this.